### PR TITLE
bdf-implementation

### PR DESCRIPTION
**Pull request by anthonyozerov**
Pull request to merge additions done this summer under the bdf-implementation branch into the master branch.

New folders include:
`bolides/data`: Various data files about particular sensor systems (e.g. FOV boundaries) and star maps for plotting.
`bolides/metadata`: Files containing attribution information for different data sources and variable descriptions.
`webapp`: Code that runs the interactive visualizer webapp. Not packaged. Nothing in the package depends on this code.
`docs`: Sphinx documentation files

The bulk of what is new lies in `bdf.py`. Other new files define helper functions and general astronomical utilities. `utils.py`, `README.rst`, `setup.py`, `__init__.py`, and `version.py` are the only files modified or deleted. All original functionality should remain.

### CODE REVIEW

File: LICENSE
Comment (jcsmithhere) [inline LICENSE:19]: We should add the MIT license but probably do not need to hold up the PR for that. We can do it later.

File: bolides/plotting.py
Comment (jcsmithhere) [inline bolides/plotting.py:7]: I tried running this and it errored due to plotly package not being available. Is this an unlisted dependency? I also presume this is experimental code because there is no documentation. If so, we should probably turn it off for the merge,

File: bolides/sdf.py
Comment (jcsmithhere) [inline bolides/sdf.py:7]: I do not have time to test this functionality. I'll just give it a pass and assume it works. It's not central to the main functionality of the bolide package.

File: docs/_templates/footer.html
Comment (jcsmithhere) [inline docs/_templates/footer.html:1]: Not necessary to complete this merge, but we should add an authors and citations page to the documentation where we list the authors in more details (affiliation, ORCID, maybe even short bio). We should also add a citation request for anyone who wants to use our package or data, and give a bibtex example of how to cite our work. That should include both the Icarus paper I wrote and some citation for your package.

File: bolides/plotting.py
Comment (jcsmithhere) [reply bolides/plotting.py:7]: Ah, I had to reinstall the package with `pip install -e .` to get `plotly` installed. OK, plot_interactive() "runs" however in my jupyter notebook it just displays a blank screen in the output cell. Should this do something?

File: webapp/README.txt
Comment (jcsmithhere) [inline webapp/README.txt:1]: I don't have time to test the webapp. I'll assume it works because it's at bolides.seti.org.

File: bolides/bdf.py
Comment (jcsmithhere) [inline bolides/bdf.py:29]: You wrote very good headers.

File: bolides/bdf.py
Comment (jcsmithhere) [inline bolides/bdf.py:193]: In the tutorial you say either use `bdf.columns` or bdf['columns']. The latter throws and error.

File: bolides/bdf.py
Comment (jcsmithhere) [inline bolides/bdf.py:331]: The detectedBy strings are 'GLM-16' and 'GLM-17' but here the filter strings are 'glm16' and 'glm17'. We should standardize the nomenclature, so should use the same strings as in the website data. We could also have short-hand options. I like 'G16' and 'G17' as short-hand.

File: bolides/bdf.py
Comment (jcsmithhere) [inline bolides/bdf.py:899]: Add a header comment for `add_website_data`. 

Also, it can take a long time to downlaod the data. Can we add a status bar ad it downloads, or at least a loading... statement?

**Review by jcsmithhere — CHANGES_REQUESTED** [review]
@anthonyozerov, This is some excellent work. You did so much this summer. I do not have the time to inspect every line and test everything you did. I did go through the tutorial and tested the main functionality. I have a bunch of comments. Some I thin kshoudl be completed before we merger but I mention others that can wait.

File: README.rst
Comment (anthonyozerov) [reply README.rst]: Slowed down to 2/3 speed in 404c8b2c6b00ee2c7d3fd50f3b22fdeed91687a9.

File: README.rst
Comment (anthonyozerov) [reply README.rst]: Removed the mention of this in 404c8b2c6b00ee2c7d3fd50f3b22fdeed91687a9. We can just not rebuild the docs on the site until the merge.

File: README.rst
Comment (anthonyozerov) [reply README.rst]: Changed to this suggestion in 404c8b2c6b00ee2c7d3fd50f3b22fdeed91687a9.

File: bolides/plotting.py
Comment (anthonyozerov) [reply bolides/plotting.py:7]: Checked on a fresh install in a new Conda environment and it seems to work. According to a comment [here](https://stackoverflow.com/questions/52771328/plotly-chart-not-showing-in-jupyter-notebook) maybe you need to completely restart the Jupyter server (not just the kernel) for plotly to work.

File: bolides/bdf.py
Comment (anthonyozerov) [reply bolides/bdf.py:194]: Clarified this by using a hypothetical column named `interesting_data` in the tutorial.

File: bolides/bdf.py
Comment (anthonyozerov) [reply bolides/bdf.py]: Implemented this in 7f49f433a12f5e91f535ef1069465bc24c20f665. The input is case-insensitive so something like 'g16' (three key-presses instead of four!) works too.

File: bolides/bdf.py
Comment (anthonyozerov) [reply bolides/bdf.py]: Added a tqdm loading bar and a header comment in 7f49f433a12f5e91f535ef1069465bc24c20f665.

File: bolides/plotting.py
Comment (jcsmithhere) [reply bolides/plotting.py:7]: Tried a fresh python environment install and `plot_interactive` now works.

File: bolides/metadata/columns.csv
Comment (jcsmithhere) [reply bolides/metadata/columns.csv]: Wording looks fine.

File: bolides/bdf.py
Comment (jcsmithhere) [reply bolides/bdf.py]: Tested, and works.

**Review by jcsmithhere — APPROVED** [review]
Changes look good!

**Comment by jcsmithhere** [comment]
Excellent work, @anthonyozerov!

**Pull request merged**
